internal/rbd/nodeserver.go

```diff
@@ -1397,8 +1406,28 @@ func getDeviceSize(ctx context.Context, devicePath string) (uint64, error) {
 }
 
 func (ns *NodeServer) SetReadAffinityMapOptions(crushLocationMap map[string]string) {
+	ns.readAffinityMapOptions = constructReadAffinityMapOption(crushLocationMap)
+}
+
+func (ns *NodeServer) getReadAffinityMapOptionFromConfigMap(rv *rbdVolume) string {
+	crushLocationLabels, err  := util.GetReadAffinityOptions(util.CsiConfigFile, rv.ClusterID)
+	if err != nil {
+		log.FatalLogMsg(err.Error())
+	}
+
+	crushLocationMap, err := util.GetCrushLocationMap(crushLocationLabels, ns.Driver.NodeID)
```

Comment:
This function call k8s api, which is an expensive operation, to get node labels.
Performance will severely be impacted if we perform such a call on each nodeStage request. 

Please fetch the node labels when the driver starts and store it.

Reply:
> Please fetch the node labels when the driver starts and store it.

- Is this a problem for a running Ceph-CSI setup when nodes are added later on?
- What happens if the Crush location map is modified?

Reply:
> > Please fetch the node labels when the driver starts and store it.
> 
> * Is this a problem for a running Ceph-CSI setup when nodes are added later on?

No, We just use this in daemonset nodeplugin pod. 
So, when a new node is added, a daemonset pod starts on it and we'll pick up the labels that time.
There is a requirement currently that when node labels are updated, they'll need to restart nodepugin pods andthe mounted volumes for the changed topology to take affect.

> * What happens if the Crush location map is modified?

If you meant Crush Location map modified by means of node topology label update, it is the same as above.
If you meant changes were done to the configmap to update labels, they'll need to restart the nodeplugin pod too.


I expect these changes to be very rare and possibly untouched after initial creation.

Reply:
I have no idea how often locations/labels are added to a crush map :man_shrugging: If it is clearly documented that changing labels of a node requires restarting, then thats fine.